CONTRIBUTING.md

```diff
@@ -1,12 +1,205 @@
+  ```sh
+  > npm run test -w wrangler
+  ```
+- Watch the files in a specific workspace (e.g. wrangler), and run the tests when anything changes
+  ```sh
+  > npm run test -w wrangler -- --watch
+  ```
+
+## Steps For Making Changes
+
+Every change you make should be stored in a [git commit](https://github.com/git-guides/git-commit).
+Changes should be committed to a new local branch, which then gets pushed to your fork of the repository on GitHub.
+
+- Ensure your `main` branch is up to date
+  ```sh
+  > git switch main
+  > git pull upstream main
+  ```
+- Create a new branch, based off the `main` branch
+  ```sh
+  > git checkout -b <new-branch-name> main
+  ```
+- Stage files to include in a commit
+  - Use [VS Code](https://code.visualstudio.com/docs/editor/versioncontrol#_git-support)
+  - Or add and commit files via the command line
+  ```sh
+  > git add <paths-to-changes-files>
+  > git commit
+  ```
+- Push changes to your fork
+  ```sh
+  git push -u origin <new-branch-name>
+  ```
+- Once you are happy with your changes, create a Pull Request on GitHub
 
 ## Changesets
 
-The project will be utilizing a workflow that will expedite and improve the ease of which contributions with changelogs done.
+Every non-trivial change to the project - those that should appear in the changelog - must be captured in a "changeset".
+We use the [`changesets`](https://github.com/changesets/changesets/blob/main/README.md) tool for creating changesets, publishing versions and updating the changelog.
+
+- Create a changeset for the current change.
+  ```sh
+  > npm run changeset
+  ```
+  or
+  ```sh
+  > npx changeset
+  ```
+- Select which workspaces are affected by the change and whether the version requires a major, minor or patch release.
+- Update the generated changeset with a description of the change.
+- Include the generate changeset in the current commit.
+  ```sh
+  > git add ./changeset/*.md
+  ```
+
+### Changeset message format
+
+Each changeset is a file that describes the change being merged. This file is used to generate the changelog when the changes are released.
+
+To help maintain consistency in the changelog, changesets should have the following format:
+
+```
+<TYPE>: <TITLE>
+
+<BODY>
+
+[BREAKING CHANGES <BREAKING_CHANGE_NOTES>]
+```
 
-- Changeset Bot
-- Create a changeset by running `npm run changeset` or `npx changeset`. [More info.](https://github.com/changesets/changesets#how-do-i-get-started)
+- `TYPE` should be a single word describing the "type" of the change. For example, one of `feature`, `fix`, `refactor`, `docs` or `chore`.
+- `TITLE` should be a single sentence containing an imperative description of the change.
+- `BODY` should be one or more paragraphs that go into more detail about the reason for the change and anything notable about the approach taken.
+- `BREAKING_CHANGE_NOTES` (optional) should be one or more paragraphs describing how this change breaks current usage and how to migrate to the new usage.
```

Comment:
This is a good set of rules. I'll start following them myself 😅

Reply:
Same 😄